### PR TITLE
Explicit subject mapping for images, metadata json serialization

### PR DESCRIPTION
**Pull request by PFWhite**
This pull request is to enable a --subid commandline arg to explicitly map all images uploaded to one subject. 

Additionally there is a new feature where image header json will be generated with the -m flag

### CODE REVIEW

File: polyjuice/dicom_image.py [hunk 1]
```diff
@@ -6,9 +10,43 @@ def __init__(self, dicom_file):
...
+                metadata[key] = self._process_data_element(val)
+            return metadata
+        else:
+            try:
+                return int(element.value)
+            except:
+                try:
+                    return float(element.value)
+                except:
+                    return str(element._value)
```
Comment (naomidb) [inline polyjuice/dicom_image.py]: 1. All 'value' or all '_value'

2. Cast as string and get rid of try/excepts

File: polyjuice/polyjuice.py [hunk 2]
```diff
@@ -88,30 +91,36 @@ def browse_restricted_section(parent_file, out_dir, zip_dir, modifications, id_p
...
                     log(failure_message)
     return
 
-def brew_potion(editor, working_file, out_dir, modifications, id_pairs, log):
-    #Use DicomCaretaker to clean files and find approprite folders to save the output
+def brew_potion(editor, working_file, out_dir, modifications, id_pairs, log, get_metadata=False):
+    """
+    Use DicomCaretaker to clean files and find approprite folders to save the output
+
+    If get_metadata is passed a file with the tags of the image will be written to the same
```
Comment (naomidb) [inline polyjuice/polyjuice.py]: "If get_metadata is passed, a file with..."

File: polyjuice/polyjuice.py [hunk 3]
```diff
@@ -124,17 +133,17 @@ def brew_potion(editor, working_file, out_dir, modifications, id_pairs, log):
...
 
-    except Exception, e:
+    except Exception as e:
         print("{} failed".format(name))
         failure_message = "{} failed".format(name) + "\n" + str(e)
         log(failure_message)
     return
 
-def add_hair(dicom_folders, zip_dir, log):
+def zip_files(dicom_folders, zip_dir, log):
```
Comment (naomidb) [inline polyjuice/polyjuice.py:146]: If you decide to keep this name for the function, the README needs to be edited.

**Review by naomidb — CHANGES_REQUESTED** [review]
Small changes

File: polyjuice/polyjuice.py (hunk 3 above)
Comment (PFWhite) [reply polyjuice/polyjuice.py:146]: readme addressed

**Pull request closed without merging**